controllers/daemonset_test.go

```diff
@@ -259,7 +263,8 @@ var _ = Describe("daemonSetGenerator", func() {
 				ObjectMeta: metav1.ObjectMeta{Name: notLegitName, Namespace: namespace},
 			}
 
-			client := fake.NewClientBuilder().WithObjects(&dsLegit, &dsNotLegit).Build()
+			client := controllers.NewMockClient(gomockCtrl)
+			client.EXPECT().Delete(context.TODO(), gomock.Any()).Return(nil)
```

Comment:
```suggestion
			client.EXPECT().Delete(context.TODO(), existingDS[1]).Return(nil)
```

Reply:
I'm not a big fan of doing it because we are coupling the tests to the implementation of the code. 
The idea behind this shouldn't be "I want you to call a specific method with specific args", instead should be "I don't care what you do, I want you to return what DS was deleted".

I don't know if I explain myself properly :joy: